### DISCUSSION TITLE
Planned changes to the Starknet docs Github repo

### DISCUSSION THREAD

**Discussion by stoobie** [starknet-io/starknet-docs#964] [Announcements]
Hi Folks,

This is Steve. I'm a technical writer at StarkWare and I maintain the Starknet docs.

I want to let you know about some minor changes I'm planning to make in the repo, essentially to streamline processes and fix some bugs.

1. **Change the Github workflows and other files as necessary to enable merging most PRs directly to `master` instead of `dev`. DONE in #961.** : Using a `dev` branch is a normal part of a source code repo, but the docs flow does not require this branch, and it adds an unnecessary step in the release and publishing flow. In a case where a PR should not be published immediately, such as a feature that is not yet released, a release branch is created to hold that content until the official release date. For example, https://github.com/starknet-io/starknet-docs/pull/933.
2. Make `master` the default branch. **DONE** (Possibly eventually remove the `dev` branch, since it is redundant.)
3. Remove auto tagging. Again, this is a workflow that makes sense in the context of a non-docs repo, such as an app, a protocol, an API, a smart contract, etc, but does not add any value in the context of this docs repo, in which changes occur much more rapidly. We do not practically speaking have multiple versions of the docs, and so tagging is not only _not_ useful, it can also be confusing for someone who expects to glean version info from any given tag.

I'm not sure about the order of the following changes.
- DONE Change the name of `master` to `main`, consistent with the default Github repo naming convention, as well as other doc-related repos in the starknet-io Github org. I think this is the right thing to do and it's something I've been meaning to do for awhile.
- [FIX] Generate previews from forked repos. This is currently broken.
- DONE [FIX] Delete PR previews from the gh-pages branch when the PR is merged and published. This is not working.
-  [FEATURE] Add a list of contributors. Ideally, this would be automatically updated. If you know how to do this, such a contribution would be very much appreciated!

**Comment by stoobie** [comment]
@ArielElp @LandauRaz @agongdai @lorcan-codes @wojciechos @odednaor @Mirko-von-Leipzig @LucasLvy @jelilat @glrn @dan-ziv @dbejarano820 @AkromDev @ptisserand @Eikix @robertkodra @tdelabro @glihm @yuvalsw @anshulforyou @roi-amaler-starkware @daizhengxue @adishildan @Brivan-26 @omahs @barretodavid @xgrizx @milancermak @thecotne @bbrandtom @nikeshnazareth @emnul @bigant88 @tudorpintea999 @zzq0826 @pandabadger @Swarag-N @AlvaroLuken @tekkac @ivpavici @tshakalekholoane 

cc: @stoobie

**Comment by omahs** [comment]
@stoobie for the Contributors list, this tool is used on many repos: https://allcontributors.org/

**Comment by stoobie** [comment]
DONE:
[FIX] Delete PR previews from the gh-pages branch when the PR is merged and published. This is not working.

**Comment by stoobie** [comment]
Update:

**DONE** Change the name of `master` to `main`, consistent with the default Github repo naming convention, as well as other doc-related repos in the starknet-io Github org.

Please follow these steps to update your local repos to use `main` instead of `master` from now on:

1. Go to the master branch: `git checkout master`
2. Rename master to main locally: `git branch -m master main`
3. Get the latest commits from the server: `git fetch`
4. Remove the link to origin/master: `git branch --unset-upstream`
5. Add a link to origin/main: `git branch -u origin/main`
6. Update the default branch to be `origin/main: git symbolic-ref refs/remotes/origin/HEAD refs/remotes/origin/main`
7. Delete origin/master completely: `git branch --remotes -d origin/master `

Source: [danieldogeanu/RenameGitBranch.md](https://gist.github.com/danieldogeanu/739f88ea5312aaa23180e162e3ae89ab)

**Comment by stoobie** [comment]
Previously, this repo used tags as part of the workflow. I removed this step as it had no particular relevance to this repo. As a result, the Changelog is no longer updated.
